src/Tools/ExternalAccess/FSharp/Internal/Formatting/FSharpFormattingInteractionService.cs

```diff
@@ -0,0 +1,54 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Composition;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.ExternalAccess.FSharp.Formatting;
+using Microsoft.CodeAnalysis.Formatting;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.CodeAnalysis.Options;
+using Microsoft.CodeAnalysis.Text;
+
+namespace Microsoft.CodeAnalysis.ExternalAccess.FSharp.Internal.Formatting
+{
+    [Shared]
+    [ExportLanguageService(typeof(IFormattingInteractionService), LanguageNames.FSharp)]
+    internal class FSharpFormattingInteractionService : IFormattingInteractionService
+    {
+        private readonly IFSharpFormattingInteractionService _service;
+
+        [ImportingConstructor]
+        [Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
+        public FSharpFormattingInteractionService(IFSharpFormattingInteractionService service)
+        {
+            _service = service;
+        }
```

Comment:
So we're defining this now, even though F# won't have the IFSharpFormattingInteractionService yet; are we relying on the fact that this will fail to MEF compose, and thus the proxy won't see the implementation of IFormattingInteractionService and fall back to the editor one? That _works_ based on MEF part rejection; that's just something we don't actually rely on often like this so wanted to confirm that'll behave right.

Reply:
That was certainly what I was hoping would happen.

Reply:
The external access layer leverages MEF part rejection heavily as a key design element.